pages/docs/couple-your-code/couple-your-code-waveform.md

```diff
@@ -0,0 +1,111 @@
+{% endnote %}
+
+The experimental API has to be activated in the configuration file via the `experimental` attribute. This allows us to define the order of the interpolant in the `read-data` tag of the corresponding `participant`. The default is constant interpolation (`waveform-order="0"`). The following example uses `waveform-order="1"` and, therefore, linear interpolation:
+
+```xml
+<solver-interface experimental="true" ... >
+...
+    <participant name="FluidSolver">
+        <use-mesh name="FluidMesh" provide="yes"/>
+        <write-data name="Forces" mesh="MyMesh"/>
+        <read-data name="Displacements" mesh="FluidMesh" waveform-order="1"/>
+    </participant>
+...
+</solver-interface>
+```
+
+## Usage example
+
+We are now ready to extend the example from ["Step 6 - Implicit coupling"](couple-your-code-implicit-coupling.html) to use waveforms. Only few changes are necessary to sample the `Displacements` at the middle of the time window, which might be necessary for our specific application:
+
+```cpp
+// no relevant changes for initialization
+// See "Step 6 - Implicit coupling"
+...
+precice_dt = precice.initialize();
+while (not simulationDone()){ // time loop
+  // write checkpoint
+  ...
+  dt = beginTimeStep(); // e.g. compute adaptive dt 
+  dt = min(precice_dt, dt);
+  if (precice.isReadDataAvailable()){ // always true, because we can sample at arbitrary points
+    // sampling in the middle of the time step
+    precice.readBlockVectorData(displID, vertexSize, vertexIDs, 0.5 * dt, displacements);
+    setDisplacements(displacements); // displacement at the middle of the time step
+  }
+  solveTimeStep(dt); // might be using midpoint rule for time-stepping
+  if (precice.isWriteDataRequired(dt)){ // only true at the end of the time window
+    computeForces(forces);
+    precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+  }
+  precice_dt = precice.advance(dt);
+  // read checkpoint & endTimeStep  
+  ...
+}
+// usual finalization
+...
+```
+
+The `precice.isReadDataAvailable()` and `precice.isWriteDataRequired(dt)` are optional, but recommended -- especially, if subcycling is used.
+
+As described in ["Step 7 - Data initialization"](couple-your-code-initializing-coupling-data), we can also use `initializeData` to provide initial data for the interpolation. If `initializeData` is not called, preCICE uses zero initial data for constructing the interpolant.
+
+## Literature
+
+[^1]: Rüth, B, Uekermann, B, Mehl, M, Birken, P, Monge, A, Bungartz, H-J. Quasi-Newton waveform iteration for partitioned surface-coupled multiphysics applications. Int J Numer Methods Eng. 2021; 122: 5236– 5257. [https://doi.org/10.1002/nme.6443](https://doi.org/10.1002/nme.6443)
```

Comment:
Have you checked how the "^1" works and looks? We don't use this on other pages, e.g. https://precice.org/tutorials-elastic-tube-1d.html

Reply:
Yes. It's basically creating a wikipedia-style footnote. I tweaked it a bit more in c9a944914147d376041886c38c88a822589dec73:
![Screenshot from 2022-02-15 11-50-05](https://user-images.githubusercontent.com/5740604/154047291-b55528a6-a53a-44b4-b0f3-a143316b1e4a.png)
I think it would also be good to use this style more often. We can open an issue on this topic.

Reply:
Could not resist: https://github.com/precice/precice.github.io/issues/130 :smirk: